ovis_demo/src/ovis_demo.cpp

```diff
@@ -0,0 +1,318 @@
+#include <algorithm>
+#include <string>
+#include <iostream>
+#include <vector>
+
+#include "ovis_demo.hpp"
+
+#include <ros/ros.h>
+#include <sensor_msgs/Joy.h>
+#include <actionlib/client/simple_action_client.h>
+#include <actionlib/client/simple_client_goal_state.h>
+#include <kinova_msgs/ArmJointAnglesAction.h>
+#include <Kinova.API.UsbCommandLayerUbuntu.h>
+
+#include <inttypes.h>
+#include <string>
+#include <stdexcept>
+#include <dlfcn.h>
+
+constexpr uint8_t NUMBER_OF_JOINTS = 6;
+constexpr uint8_t NUMBER_OF_DEGREE_PER_GOAL = 1;
+
+static uint8_t joint_index;
+static uint8_t joint_active[NUMBER_OF_JOINTS];
+static bool bumper_left_pressed = false;
+static bool bumper_right_pressed = false;
+static AngularPosition joints_angles;
+static TrajectoryPoint trajectory_point;
+
+static unsigned int const LEFT_BUMPER_INDEX = 4;
+static unsigned int const RIGHT_BUMPER_INDEX = 5;
+
+void sendJointCommand(std::vector<float> const joy_axes)
+{
+  // Act as a deadzone for the gamepad
+  // TODO remove and use launch parameter
+  if (joy_axes[1] < 0.1 && joy_axes[1] > -0.1)
+  {
+    return;
+  }
+
+  int direction = (joy_axes[1] > 0.0) ? 1 : -1;
+
+  int result = 0;
+  result = getAngularPosition(joints_angles);
+  if (result != NO_ERROR_KINOVA)
+  {
+    ROS_ERROR("Could not get the angular position. Result = [%d]", result);
+  }
+
+  switch (joint_index)
```

Comment:
Some actuators are physically set up differently than other in the arm. What happen is that when moving in a certain direction for one actuator when switching for another one. The direction will be inverted. 

So what are the option : 
- Use ros param to inverse the direction for certain actuator (like for the flippers)
- Try to reconfigure them in the code ? 
- ???